src/main/scala/org/broadinstitute/dsde/firecloud/webservice/UserApiService.scala

```diff
@@ -83,43 +83,47 @@ trait UserApiService extends HttpService with PerRequestCreator with FireCloudRe
-                  // rawls 404 means the user is not registered with FireCloud
-                  case NotFound => respondWithErrorReport(NotFound, "FireCloud user registration not found", requestContext)
-                  // rawls error? boo. All we can do is respond with an error.
-                  case InternalServerError => respondWithErrorReport(InternalServerError, InternalServerError.defaultMessage, requestContext)
-                  // rawls found the user; we'll try to parse the response and inspect it
+                  // Sam rejected our request. User is either invalid or their token timed out; this is truly unauthorized
+                  case Unauthorized =>
+                    respondWithErrorReport(Unauthorized, "Request rejected by identity service - invalid user or expired token.", requestContext)
+                  // Sam 404 means the user is not registered with FireCloud
+                  case NotFound =>
+                    respondWithErrorReport(NotFound, "FireCloud user registration not found.", requestContext)
+                  // Sam error? boo. All we can do is respond with an error.
+                  case InternalServerError =>
+                    respondWithErrorReport(InternalServerError, "Identity service encountered an unknown error, please try again.", requestContext)
+                  // Sam found the user; we'll try to parse the response and inspect it
                   case OK =>
                     val respJson = response.entity.as[RegistrationInfo]
                     respJson match {
                       case Right(regInfo) =>
                         if (regInfo.enabled.google && regInfo.enabled.ldap && regInfo.enabled.allUsersGroup) {
-                          // rawls says the user is fully registered and activated!
+                          // Sam says the user is fully registered and activated!
                           requestContext.complete(OK, regInfo)
                         } else {
-                          // rawls knows about the user, but the user isn't activated
-                          respondWithErrorReport(Forbidden, "FireCloud user not activated", requestContext)
+                          // Sam knows about the user, but the user isn't activated
+                          respondWithErrorReport(Forbidden, "FireCloud user not activated.", requestContext)
                         }
-                      // we couldn't parse the rawls response. Respond with an error.
-                      case Left(error) =>
-                        respondWithErrorReport(InternalServerError, InternalServerError.defaultMessage, requestContext)
+                      // we couldn't parse the Sam response. Respond with an error.
+                      case Left(_) =>
+                        // TODO: no obvious way to include the JSON parsing error in the ErrorReport. We define `message` below and it does not belong there.
```

Comment:
I think our best bet is to add a `description` field to `ErrorReport` and add support for it to UI's `ErrorViewer`

Reply:
It's pretty ugly, but you could do something like:
```
                      // we couldn't parse the Sam response. Respond with an error.
                      case Left(e) =>
                        val er:ErrorReport = e match {
                          case MalformedContent(errorMessage: String, None) => ErrorReport(errorMessage)
                          case MalformedContent(errorMessage: String, Some(cause:Throwable)) => ErrorReport(InternalServerError, errorMessage, cause)
                          case UnsupportedContentType(errorMessage: String) => ErrorReport(errorMessage)
                          case _ => ErrorReport("Deserialization error.")
                        }
                        respondWithErrorReport(InternalServerError, s"Received unparseable response from identity service: ${er.message}", er, requestContext)

...
...

private def respondWithErrorReport(statusCode: StatusCode, message: String, cause: ErrorReport, requestContext: RequestContext): Unit = {
    requestContext.complete(statusCode, ErrorReport(statusCode = statusCode, message = message, cause = cause))
  }

```